8thwall-camera.js

```diff
@@ -5,88 +5,402 @@
-        this.camera = vals[this.camera];
-        if(this.camera == 'auto') {
-            this.camera = 'back';
+                // User had no interaction with the page so far
+                if (exception.name === "NotAllowedError") {
+                    const motionEvent = await this.promptForDeviceMotion();
+                    if (motionEvent !== "granted") {
+                        throw new Error("MotionEvent");
+                    }
+                } else {
+                    throw new Error("MotionEvent");
+                }
+            }
+        }
+
+        try {
+            // make sure we get the camera stream
+            const stream = await navigator.mediaDevices.getUserMedia({ video: true, audio: true })
+
+            // If we successfully acquired the camera stream - we can stop it and wait until 8thwall requests it again
+            stream.getTracks().forEach((track) => {
+                track.stop();
+            });
+
+        } catch (exception) {
+            throw new Error("Camera");
         }
+    },
+
 
-        this.onStart = this.onStart.bind(this);
+    init: function () {
+        this.view = this.object.getComponent("view");
         this.onUpdate = this.onUpdate.bind(this);
+        this.onAttach = this.onAttach.bind(this);
+        this.onException = this.onException.bind(this);
+        this.onCameraStatusChange = this.onCameraStatusChange.bind(this);
+    },
+
+    start: async function () {
+        console.log("useCustomUIOverlays", this.useCustomUIOverlays);
+        if (!this.useCustomUIOverlays) {
+            console.log("Initing")
+            OverlaysHandler.init();
+        }
 
+        try {
+            await this.getPermissions();
+        } catch (error) {
+            // User did not grant the camera or motionEvent permissions
+            window.dispatchEvent(new CustomEvent("8thwall-permission-fail", {detail: error}))
+            return;
+        }
+
+        await waitForXR8();
+
+        XR8.XrController.configure({
+            // enableLighting: true,
+            disableWorldTracking: false
+        });
+
+        this.GlTextureRenderer = XR8.GlTextureRenderer.pipelineModule();
         XR8.addCameraPipelineModules([
-            /* Draw the camera feed */
-            XR8.GlTextureRenderer.pipelineModule(),
-            XR8.XrController.pipelineModule(),
-            this,
+            this.GlTextureRenderer, // Draws the camera feed.
+            XR8.XrController.pipelineModule(), // Enables SLAM tracking.
+            this
         ]);
 
-        if(this.camera == 'back') {
-            XR8.run({canvas: Module['canvas'], ownRunLoop: false});
+        const config = {
+            cameraConfig: {
+                direction: XR8.XrConfig.camera().BACK,
+            },
+            canvas: Module.canvas,
+            allowedDevices: XR8.XrConfig.device().ANY,
+            ownRunLoop: false,
+        }
+        XR8.run(config);
+    },
+
+    /**
+     * @param {*} params 
```

Comment:
The param doc here is not too useful :)

If it's private, is there a way to prepend `_`? That's usually how we mark private members and functions.